controllers/topology/reconcile_state.go

```diff
@@ -374,10 +385,63 @@ func (r *ClusterReconciler) reconcileReferencedTemplate(ctx context.Context, in
 	// Set up a cleanup func for removing the old template.
 	// NOTE: This function must be called after updating the object containing the reference to the Template.
 	return func() error {
-		log.Info("Deleting", in.desired.GroupVersionKind().String(), in.desired.GetName())
+		logReconcileAction(log, deleteObjectReconcileOperation, in.current)
 		if err := r.Client.Delete(ctx, in.current); err != nil {
 			return errors.Wrapf(err, "failed to delete %s/%s", in.desired.GroupVersionKind(), in.desired.GetName())
 		}
 		return nil
 	}, nil
 }
+
+type reconcileAction int
+
+const (
+	createObjectReconcileOperation reconcileAction = iota
+	noChangesReconcileOperation
+	patchObjectReconcileOperation
+	rotateTemplateReconcileOperation
+	deleteObjectReconcileOperation
+)
+
+// logReconcileAction takes care of creating consistent log message for reconcile state.
+// NOTE: In order to improve log readability for people using kubectl logs, we are adding
+// a minimal set of information about the object being reconciled to the message as well.
+func logReconcileAction(log logr.Logger, op reconcileAction, obj client.Object, keyAndValues ...interface{}) {
+	// Add default keyAndValues providing info about the object
+	keyAndValues = append(keyAndValues,
+		"object GroupVersion", obj.GetObjectKind().GroupVersionKind().GroupVersion().String(),
+		"object Kind", obj.GetObjectKind().GroupVersionKind().Kind,
+		"object", obj.GetName(),
```

Comment:
```suggestion
		"object groupVersion", obj.GetObjectKind().GroupVersionKind().GroupVersion().String(),
		"object kind", obj.GetObjectKind().GroupVersionKind().Kind,
		"object", obj.GetName(),
```
nit, if you want.
Based on the controller-runtime keys (e.g. `reconciler kind`) it would look more consistent that way. 
```bash
I0820 18:26:01.101400      35 reconcile_state.go:303] controller-runtime/manager/controller/cluster "msg"="Creating infrastructure.cluster.x-k8s.io/v1alpha4, Kind=DockerCluster/my-cluster1-rl4sf" "Reconciler"="topology/ClusterReconciler" "name"="my-cluster1" "namespace"="default" "reconciler group"="cluster.x-k8s.io" "reconciler kind"="Cluster" 
I0820 18:26:01.113888      35 reconcile_state.go:353] controller-runtime/manager/controller/cluster "msg"="Creating infrastructure.cluster.x-k8s.io/v1alpha4, Kind=DockerMachineTemplate/my-cluster1-controlplane-gtwbq" 
```

Reply:
Same below in loggerWithMachineDeployment and at least in l.374